src/common/com/intellij/plugins/haxe/model/type/HaxeGenericResolverUtil.java

```diff
@@ -177,4 +181,18 @@ public static HaxeGenericResolver generateResolverForSupers(HaxeClass clazz) {
     return resolver;
   }
 
+  // Hack?
+  // Since null<T> references in some places are handled as if they where type T
+  // we also have to support resolving Type Parameters as if Null<T> was just T
```

Comment:
I get what you're doing here.  My question (without doing any research) is whether we can treat Null<T> specially in the resolver, so that we don't have to write lots of code to deal with the Null all over the place?  This isn't the first place we're handling it specially, and it won't be the last.  I'd like it better if we could deal with it once and for all.  I'm not sure if we can do that, though.

I'm not rejecting this change at all -- or even suggesting that you go do the research.  Just wondering if it can be handled better.